operator/node.go

```diff
@@ -111,7 +111,7 @@ func New(logger *zap.Logger, opts Options, slotTickerProvider slotticker.Provide
 			Network:            opts.Network,
 			ShareStorage:       opts.ValidatorOptions.RegistryStorage.Shares(),
 			RecipientStorage:   opts.ValidatorOptions.RegistryStorage,
-			OperatorData:       opts.ValidatorOptions.OperatorData,
+			GetOperatorID:      func() spectypes.OperatorID { return opts.ValidatorController.GetOperatorData().ID },
```

Comment:
can we maybe try to add GetOperatorID function to `ValidatorController` instead of creating this func here and elsewhere?

then we can just pass `opts.ValidatorController.GetOperatorID``

or require an `OperatorIDProvider` interface and pass `opts.ValidatorController` directly

Reply:
@moshe-blox done, please review again